packages/thirdweb/src/extensions/erc1155/read/getOwnedTokenIds.ts

```diff
@@ -0,0 +1,90 @@
+import type { Address } from "abitype";
+import type { BaseTransactionOptions } from "../../../transaction/types.js";
+import { balanceOfBatch } from "../__generated__/IERC1155/read/balanceOfBatch.js";
+import { nextTokenIdToMint } from "../__generated__/IERC1155Enumerable/read/nextTokenIdToMint.js";
+import { nextTokenId } from "../__generated__/Zora1155/read/nextTokenId.js";
+
+const DEFAULT_QUERY_ALL_COUNT = 100;
+
+/**
+ * Parameters for retrieving owned tokenIds of an ERC1155 contract.
+ * @extension ERC1155
+ */
+export type GetOwnedTokenIdsParams = {
+  /**
+   * Which tokenId to start at.
+   */
+  start?: number;
+  /**
+   * The number of NFTs to retrieve.
+   */
+  count?: number;
+  /**
+   * The address of the wallet to get the NFTs of.
+   */
+  address: string;
+};
+
+/**
+ * Retrieves the owned ERC1155 tokenIds & the owned balance of each tokenId for a given wallet address.
+ * @param options - The transaction options and parameters.
+ * @returns A promise that resolves to an array of ERC1155 NFTs owned by the wallet address, along with the quantity owned.
+ * @extension ERC1155
+ * @example
+ * ```ts
+ * import { getOwnedNFTs } from "thirdweb/extensions/erc1155";
+ * const ownedTokenIds = await getOwnedTokenIds({
+ *  contract,
+ *  start: 0,
+ *  count: 10,
+ *  address: "0x123...",
+ * });
+ * ```
+ */
+export async function getOwnedTokenIds(
```

Comment:
should we export this? what did you need it for

Reply:
Let's say I need a list of owned tokenids for rendering a list of <NFT.Media />. Using getOwnedNFTs is a waste of requests, so I split the code from the current getOwnedNFTs to make a new function

Reply:
Makes sense. Just make sure to add it to the exports

Reply:
Once we connect the SDK to the indexer it'll be equally as costly to get the whole NFT vs just the IDs. If we export getOwnedTokenIds now, we'll later have to support two functions that practically do the same thing. So I would actually lean towards not exporting this so we can remove it once we have the indexer.

Reply:
We'll still need a fallback tho right? Fine either way, we can export it later.

I think we have it split for 721 too

Reply:
We do have it split for ERC721. I like splitting it. But I'll let you guys decide on this one

Reply:
@gregfromstl on second thought, I'm not sure what you mean by having to support 2 functions if we decide to split the code. We can totally have this new one as an internal function if you want.

I only see the benefits of this PR are: code is more modular, easier to write tests and more flexible

Reply:
@gregfromstl lets just merge it - i think its useful even with indexing to not get all metadata at once